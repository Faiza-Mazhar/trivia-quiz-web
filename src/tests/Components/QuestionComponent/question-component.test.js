import QuestionFormComponent from "../../../Components/QuestionComponent/question-component";
import renderer from "react-test-renderer";

const questionData = {
  answers: ["answer1", "answer2", "answer3", "answer4"],
  category: "Category 1",
  correctAnswer: "answer1",
  difficulty: "hard",
  question: "Question 1 Statement",
  type: "multiple",
};

test("Question component shows the questions and submit button if showSubmitButton set as true", () => {
  const component = renderer.create(
    <QuestionFormComponent
      question={questionData.question}
      answers={questionData.answers}
      handleSubmit={jest.fn()}
      handleChange={jest.fn()}
      showSubmitButton={true}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Question component to show with submit button set as false", () => {
  const component = renderer.create(
    <QuestionFormComponent
      question={questionData.question}
      answers={questionData.answers}
      handleSubmit={jest.fn()}
      handleChange={jest.fn()}
      showSubmitButton={false}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
