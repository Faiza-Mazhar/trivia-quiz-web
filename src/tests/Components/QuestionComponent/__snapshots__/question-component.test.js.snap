// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question component to show with submit button 1`] = `
<div>
  <div
    className="question"
  >
    Question 1 Statement
  </div>
  <form
    onSubmit={[MockFunction]}
  >
    <div>
      <div
        className="answers-container"
      >
        <div
          className="answer-container"
        >
          <label
            className="radio-button-label"
          >
            answer1
            <input
              name="answers"
              onClick={[Function]}
              type="radio"
              value="answer1"
            />
          </label>
        </div>
        <div
          className="answer-container"
        >
          <label
            className="radio-button-label"
          >
            answer2
            <input
              name="answers"
              onClick={[Function]}
              type="radio"
              value="answer2"
            />
          </label>
        </div>
        <div
          className="answer-container"
        >
          <label
            className="radio-button-label"
          >
            answer3
            <input
              name="answers"
              onClick={[Function]}
              type="radio"
              value="answer3"
            />
          </label>
        </div>
        <div
          className="answer-container"
        >
          <label
            className="radio-button-label"
          >
            answer4
            <input
              name="answers"
              onClick={[Function]}
              type="radio"
              value="answer4"
            />
          </label>
        </div>
      </div>
      <div
        className="submit-button"
      >
        <button
          className=" custom-button"
          type="submit"
        >
          SUBMIT
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`Question component to show with submit button set as false 1`] = `
<div>
  <div
    className="question"
  >
    Question 1 Statement
  </div>
  <form
    onSubmit={[MockFunction]}
  />
</div>
`;
